File: Java/JSONParser.java
/*
 * jsonFASTParse
 * @author: Brandon Milton
 * http://brandonio21.com
 *
 * A small Java library meant to parse JSON in a quick fashion, given that the
 * end-user is aware of the JSON data schema. Does not support dictionaries.
 */

class JSONParser {

  public static JSONObject getObjectFromString(String str) {
    // Check if surrounded by quotes, list brackets, or nothing
    if (str.charAt(0) == '[' && str.charAt(str.length() - 1) == ']')
      return new JSONList(str.substring(1, str.length() - 1));
    else if (str.charAt(0) == '"' && str.charAt(str.length() - 1) == '"')
      return new JSONObject(str.substring(1, str.length() - 1));
    else
      return new JSONObject(str);
  }

}

File: C++/json_parser.cpp
#include "json_parser.hpp"
#include "json_list.hpp"

JSONObject* JSONParser::get_obj_from_str(std::string str) {
    // Check if surrounded by quotes, list brackets, or nothing
    if (str[0] == '[' && str[str.length() - 1] == ']')
      return new JSONList(str.substr(1, str.length() - 2));
    else if (str[0] == '"' && str[str.length() - 1] == '"')
      return new JSONObject(str.substr(1, str.length() - 2));
    else
      return new JSONObject(str);
}
